spec/css/selector.hrx

```diff
@@ -359,3 +359,119 @@ Error: unterminated attribute selector for a
 >> [a=b ï] {c: d}
 
    ---^
+
+<===>
+================================================================================
+<===> escaping/number_as_first_char_without_space/input.scss
+.\31u {clear: left;}
+
+<===> escaping/number_as_first_char_without_space/output.css
+.\31 u {
+  clear: left;
+}
+
+<===> escaping/number_as_first_char_without_space/output-libsass.css
+.\31u {
+  clear: left;
+}
+
+<===>
+================================================================================
+<===> escaping/number_as_first_char_with_space/input.scss
+.\31 u {clear: left;}
+
+<===> escaping/number_as_first_char_with_space/output.css
+.\31 u {
+  clear: left;
+}
+
+<===>
+================================================================================
+<===> escaping/number_as_nonfirst_char_without_space/input.scss
+.a\31u {clear: left;}
+
+<===> escaping/number_as_nonfirst_char_without_space/output.css
+.a1u {
+  clear: left;
+}
+
+<===> escaping/number_as_nonfirst_char_without_space/output-libsass.css
+.a\31u {
+  clear: left;
+}
+
+<===>
+================================================================================
+<===> escaping/number_as_nonfirst_char_with_space/input.scss
+.a\31 u {clear: left;}
+
+<===> escaping/number_as_nonfirst_char_with_space/output.css
+.a1u {
+  clear: left;
+}
+
+<===> escaping/number_as_nonfirst_char_with_space/output-libsass.css
+.a\31 u {
+  clear: left;
+}
+
+<===>
+================================================================================
+<===> escaping/dollar_char/input.scss
+.u\$ {clear: left;}
+
+<===> escaping/dollar_char/output.css
+.u\$ {
+  clear: left;
+}
+
+<===>
+================================================================================
+<===> escaping/dollar_char_as_numeric/input.scss
+.u\24 {clear: left;}
+
+<===> escaping/dollar_char_as_numeric/output.css
+.u\$ {
+  clear: left;
+}
+
+<===> escaping/dollar_char_as_numeric/output-libsass.css
+.u\24 {
+  clear: left;
+}
+
+<===>
+================================================================================
+<===> escaping/parenthesis_in_interpolation/input.scss
```

Comment:
It's unclear to me what [single behavior](https://github.com/sass/sass-spec/blob/master/STYLE_GUIDE.md#do-test-only-one-thing-per-spec) this spec is testing. Is it non-identifier characters in interpolation? If so, why it the interpolation so complex to generate? Why not just write `.u#{"$"}`?